firestore.rules

```diff
@@ -47,7 +51,26 @@ service cloud.firestore {
         string(request.auth.token.platform_user_id) == resource.data.uid;
     }
 
-    function classInClasses() {
+    function classInRequestClasses() {
+      return isAuthed() && request.auth.token.class_hash in request.resource.data.classes;
```

Comment:
If I'm reading this right, it means a user could update a support document with other people's classes. As long as the user also includes their own class in the list.   So any user could add supports to any other class.

But I'd guess this isn't possible to prevent if you need multiple class support since the token doesn't include all of the possible classes a teacher has access too.

Reply:
The feature exists to allow teachers to share supports among all of their classes. Client-side, we figure out all of the teacher's compatible classes (i.e. classes that have assigned the same problem) and so the list of classes should be a list of the current teacher's compatible classes. As you suggest, however, we don't have a way to confirm that via server-authenticated tokens so validating that the teacher is associated with at least one of the listed classes seems like the best we can do as far as class validation is concerned.

That said, for `update` and `delete` we also include the `resourceMatchUserId()` test which limits teachers to only updating/deleting their own supports. It looks like I should add tests for those cases as well.